### ISSUE TITLE
fluctuating test accuracy

### ISSUE THREAD

**Issue by puerrrr** [HKUNLP/diffusion-of-thoughts#2]
Thanks for publishing your great work. When I train on the 4by4 multiplication dataset, even though the loss seems to converge quite well, the test accuracy fluactuates a lot. Was this expected? The settings I used was: python train.py --digit --fix_src --dataset 4by4 --steps 120000 --weights_path plaid1b_weights --min_prob 0.95

![test_acc](https://github.com/user-attachments/assets/722f0c28-a1ff-4d7c-8f6d-91fd109f3c52)
![val_nll](https://github.com/user-attachments/assets/39e2668f-f59d-454b-8aa6-e203549cbb76)

**Comment by summmeer** [comment]
This test acc curve is just for reference but maybe not accurate. Did you evaluate the final model and also get the very low acc?

**Comment by jiacheng-ye** [comment]
I just searched the previous log, and here is an experiment on the 5x5 task with 60k training steps for your reference. 4x4 should be similar as the task is easier.
![test_acc](https://github.com/user-attachments/assets/8366b2e8-28db-4e12-b30e-ffdc62669529)
![val_nll](https://github.com/user-attachments/assets/4109f708-819c-4d2c-a2b3-5dd6093bb3f7)

**Comment by puerrrr** [comment]
Thanks for replying. I got similar loss curve but accuracy was unstable. I evaluated the final model and able to get around 99% accuracy. I wonder if this was because the sampling_timesteps was set to 16 during training but 64 when evalutating?

**Comment by summmeer** [comment]
@puerrrr Yes, it is possible. Like I was saying this curve may be not accurate.

**Comment by puerrrr** [comment]
Thanks for the clarification!
